tmt/uuid.py

```diff
@@ -0,0 +1,112 @@
+import re
+from uuid import uuid4
+
+import fmf
+
+import tmt.base
+
+from .utils import log
+
+ID_FMF_ITEM = "id"
+
+
+class IDError(Exception):
+    pass
+
+
+class IDLeafError(IDError):
+    pass
+
+
+def dereference_node(node):
+    """
+    find fmf items inside node
+    """
+    if isinstance(node, fmf.Tree):
+        return node
+    elif isinstance(node, tmt.base.Node):
+        return node.node
+    else:
+        raise ValueError(f"{node} is not part of fmf data")
+
+
+def id_location(node, name=ID_FMF_ITEM):
```

Comment:
What's the point of `name=ID_FMF_ITEM`? Other functions are not providing a choice and support `ID_FMF_ITEM` only. I'd use is consistently across all functions. Plus `name` is quite confusing when reading the code.